test/acvp_test.c

```diff
@@ -1693,6 +1708,54 @@ static int aes_cfb1_bits_test(void)
     return ret;
 }
 
+static
+OSSL_PARAM *construct_snmpkdf_params(char *digest, unsigned char *eid, size_t eid_len,
+                                     unsigned char *password, size_t password_len)
+
+{
+    OSSL_PARAM *p = OPENSSL_malloc(sizeof(OSSL_PARAM) * 4);
+    OSSL_PARAM *p_ret = p;
+
+    if (p == NULL)
+        return NULL;
+
+    *p++ = OSSL_PARAM_construct_utf8_string(OSSL_KDF_PARAM_DIGEST, digest, 0);
+    *p++ = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_SNMPKDF_EID,
+                                             eid, eid_len);
+    *p++ = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_PASSWORD, password,
+                                             password_len);
+    *p = OSSL_PARAM_construct_end();
+
+    return p_ret;
+}
+
+static int test_kdf_snmpkdf(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+    static unsigned char EID1[] = {
+        0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    static unsigned char expected1[] = {
+        0x62, 0x98, 0x10, 0xa3, 0x1a, 0xdc, 0x31, 0x76, 0x9d, 0xfe, 0xbf, 0x9b,
+        0xac, 0xba, 0x06, 0xa3, 0xff, 0xd9, 0xc4, 0x9f };
+    static unsigned char password1[] = {'L', 'P', 'U', 'Q', 'D', 'L', 's', 'K' };
```

Comment:
Ought these be hex constants?  For machines which don't use ASCII.

Reply:
acvp_test is a demonstration of how to implement FIPS functional testing.  It's not meant to be a complete test.

@slontis is right, there is no need to include it.  If it needed some special handling or did something a bit weird, then it should be included.  Seeing that it doesn't, there is no need to use it as an example.